### PR TITLE
Containerize PaSe into a pair of Docker containers

### PR DESCRIPTION
**Pull request by oSoMoN**
(pase-indexer and pase-server)
to make it easier to build and run locally.

### CODE REVIEW

**Comment by oSoMoN** [comment]
Note that I dropped the "@testing-library/…" dependencies because they were creating dependency problems when building inside a container, and because they were unused. Of course they would need to be added back (and the dependency problems investigated further) if tests were to be written for the frontend.
I took the opportunity to bump the react dependencies to recent versions.

File: README.md [hunk 1]
```diff
@@ -23,4 +23,16 @@ Once indexing has finished, you can search:
   - use `java -jar pase.jar serve <index_path>` to start the PaSe Server
   - visit [http://localhost:4567](http://localhost:4567) with your browser
 
-![Screen capture of PaSe's Web UI](doc/pase-webui.gif)
\ No newline at end of file
+![Screen capture of PaSe's Web UI](doc/pase-webui.gif)
+
+## Building and running as Docker containers
```
Comment (moio) [inline README.md]: Please move the "Building" part of this text to CONTRIBUTING.md, leaving only end user instructions here.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
+FROM maven:3-eclipse-temurin-11-alpine AS builder
```
Comment (moio) [inline Dockerfile]: I am fine with this choice, although I can recommend [SLE BCI](https://registry.suse.com/) images as well, as they are maintained by the company.

Not blocking for this PR anyway.

**Review by moio — APPROVED** [review]
Thanks for this contribution! I left a couple of comments, anyway those changes are fine.

I will let @boiko the honor of merging 😉

File: README.md (hunk 1 above)
Comment (oSoMoN) [reply README.md]: Done.

File: Dockerfile (hunk 2 above)
Comment (oSoMoN) [reply Dockerfile]: Done, thanks for the suggestion!
The resulting images are slightly larger than when using alpine as a base (335MB vs 307MB), but that looks reasonable.

**Pull request merged**